src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EvpPkey.cs

```diff
@@ -274,6 +260,28 @@ internal static ArraySegment<byte> RentEncodeSubjectPublicKeyInfo(SafeEvpPKeyHan
             return pkey;
         }
 
+        [LibraryImport(Libraries.CryptoNative, StringMarshalling = StringMarshalling.Utf8)]
+        private static partial IntPtr CryptoNative_LoadKeyFromProvider(
+            string providerName,
+            string keyUri,
+            ref IntPtr extraHandle);
+
+        internal static SafeEvpPKeyHandle LoadKeyFromProvider(
+            string providerName,
+            string keyUri)
+        {
+            IntPtr extraHandle = IntPtr.Zero;
+            IntPtr evpPKeyHandle = CryptoNative_LoadKeyFromProvider(providerName, keyUri, ref extraHandle);
```

Comment:
they are really bound together because I changed the EvpPkeyDestroy and UpRefEvpPkey to take two arguments so it's not really possible to free one and not the other

Reply:
Amusingly, this can leak... under low memory conditions.  Specifically, the P/Invoke succeeds, but then `new SafeEvpPKeyHandle` throws an OutOfMemoryException.

If the P/Invoke were declared to return SafeEvpPKeyHandle then the P/Invoke stub will defend against that situation for you; you'd just need to set `ExtraHandle` after construction.  If you are opposed to setting it after construction, then this should look like

```C#
IntPtr evpPKeyHandle = IntPtr.Zero;
IntPtr extraHandle = IntPtr.Zero;

try
{
    evpPKeyHandle = CryptoNative_LoadKeyFromProvider(providerName, keyUri, ref extraHandle);

    if (evpPKeyHandle == IntPtr.Zero)
    {
        throw something;
    }

    return new SafeEvpPKeyHandle(evpPKeyHandle, extraHandle);
}
catch
{
    if (evpPKeyHandle != IntPtr.Zero)
    {
        WhateverTheFreeHandleIs(evpPKeyHandle, extraHandle);
    }

    throw;
}
```

Reply:
I will change this but OOM is killing the process which means everything gets freed anyway or am I missing something?